docs/operations/advanced/common-errors.md

```diff
@@ -0,0 +1,492 @@
+     nc -zv party1-external.kms-threshold.svc.cluster.local 50001
+   ```
+
+**Prevention:**
+- Ensure stable network connectivity between threshold parties
+- Monitor party health and restart lagging nodes
+- Consider increasing timeout values in high-latency environments
+
+---
+
+### TFHE Version Mismatch
+
+**Error Messages:**
+```bash
+assertion `left == right` failed
+  left: LweSize(1281)
+ right: LweSize(1025)
+# or
+We can't deserialize our own validated sks key: DeserializationError("invalid value: integer `1`, expected variant index 0 <= i < 1")
+```
+
+**Root Cause:** Incompatible TFHE-rs versions between components or using keys from different versions.
+
+**Solutions:**
+```bash
+# 1. Check TFHE versions in use
+grep tfhe Cargo.toml
+
+# 2. Regenerate keys with current version
+rm -rf keys/
+# Run key generation process
+
+# 3. For coprocessor version mismatch, update image:
+# In docker-compose.yaml:
+# coproc:
+#   image: ghcr.io/zama-ai/fhevm-coprocessor:v0.1.0-3  # Uses tfhe-rs 0.9
+```
+
+**Prevention:**
+- Use consistent TFHE-rs versions across all components
+- Test key compatibility after version upgrades
+- Backup keys before version changes
+
+---
+
+## Configuration Errors
+
+### Invalid Configuration
+
+**Error Message:**
+```bash
+Config validation failed: ...
+```
+
+**Diagnosis:**
+```bash
+# Validate configuration
+kms-health-check config --file config.toml
+
+# Check specific validation errors
+cargo run --bin kms-server -- --config config.toml --dry-run
+```
+
+**Common Issues:**
+- **Threshold too high:** For 13-party network, threshold=4 requires minimum 5 parties online
+- **Invalid addresses:** Check IP addresses and ports in peer configuration
+- **Missing TLS certificates:** Ensure certificate paths are correct and accessible
+- **Storage configuration:** Verify storage backend settings and permissions
+
+---
+
+### Wrong Command Arguments
+
+**Error Messages:**
+```bash
+error: unexpected argument '--pub-url' found
+error: unexpected argument '--priv-url' found
+```
+
+**Solution:** Update to current argument format:
+```bash
+# Old (deprecated):
+--pub-url file:///path/to/keys
+--priv-url file:///path/to/keys
+
+# New (current):
+--pub-path /path/to/keys
+--priv-path /path/to/keys
+```
+
+---
+
+## Docker & Container Issues
+
+### Docker Build Failures
+
+**Error Messages:**
+```bash
+pull access denied for tfhe-core, repository does not exist or may require 'docker login'
+# or
+Error loading from ghcr.io during docker image generation
+```
+
+**Solutions:**
+```bash
+# 1. Login to GitHub Container Registry
+echo '<your-github-token>' | docker login ghcr.io -u <username> --password-stdin
+
+# 2. For Chainguard images, login with proper token:
+# Visit: https://console.chainguard.dev/org/zama.ai/settings/pull-tokens
+# Use your Zama Google account (not GitHub)
+
+# 3. Ensure token has proper permissions:
+# - read:packages for GitHub
+# - Zama organization access for Chainguard
+```
+
+---
```

Comment:
Should we also add this possible error case here, or do we assume something like this will nenve happen after deployment?
https://www.notion.so/zamaai/Encountered-issues-1002797be84a470ea093930bc6c971a1?source=copy_link#26f5a7358d5e8094be01cbd2d39771e3

Reply:
Don't we have that already there? https://www.notion.so/zamaai/Encountered-issues-1002797be84a470ea093930bc6c971a1?source=copy_link#26f5a7358d5e8094be01cbd2d39771e3